components/icon/Icon.razor.cs

```diff
@@ -109,6 +113,10 @@ protected virtual async Task SetupSvgImg()
             {
                 var svg = IconService.GetIconImg(Type.ToLowerInvariant(), Theme.ToLowerInvariant());
                 _svgImg = IconService.GetStyledSvg(svg, svgClass, Width, Height, Fill, Rotate);
+                if (Theme == IconThemeType.Twotone)
+                {
+                    _svgImg = await IconService.GetTwotoneSvgIcon(_svgImg, TwotoneColor);
```

Comment:
On the other hand, JS Interop calls need to be made after ensuring that they have been rendered. So it cannot be called during the prerendering phase.

Reply:
so, what do you think? move `if (Theme == IconThemeType.Twotone)` block to after render?

Reply:
I think we can pass a flag `rendered` into SetupSvgImg method. Call it in OnInitializedAsync with `rendered=false`, and in `OnAfterRenderAsync`  with   `rendered=true` after first render.

```
SetupSvgImg(bool rendered)
```

Reply:
add a flag `_hasRendeered`,   `if (_hasRendered && Theme == IconThemeType.Twotone)`

Reply:
I found that doing so caused a flash, showing the gray first and then the color would render twice.
![icon-two-tone](https://user-images.githubusercontent.com/7550366/173985115-2dd0dd2d-d369-4d66-b81e-5c2b472b4408.gif)

![icon-two-tone](https://user-images.githubusercontent.com/7550366/173983983-bdc52591-6903-469c-b7d2-503afbfb900a.gif)

Reply:
that because it render(grey) in prerender stage, and then re-render(twotone) in afterrender stage again.
If do not want this, I think the only choice is not render in prerender stage, it will cause another problem, it will display until after-render stage. 
![ff1](https://user-images.githubusercontent.com/1829209/174000234-c2410e4c-3d76-4bd6-9006-18aa0675cbf9.gif)
![ff2](https://user-images.githubusercontent.com/1829209/174000262-bf54d006-8f5d-401d-9230-25252e8f11ce.gif)